com.unity.sg2/Editor/GraphDeltaRegistry/Registry/MultiplyNode.cs

```diff
@@ -2,10 +2,38 @@
 using UnityEditor.ShaderGraph.GraphDelta;
 using System.Linq;
 using UnityEngine;
+using UnityEditor.ShaderGraph.Defs;
 
 namespace UnityEditor.ShaderGraph.GraphDelta
 {
 
+    internal class MultiplyNodeUI : INodeUIDescriptorBuilder
+    {
+        public NodeUIDescriptor CreateDescriptor(NodeHandler node)
+        {
+            var key = node.GetRegistryKey();
+
+            bool isMatrixMultiplication =
+            GraphTypeHelpers.GetHeight(node.GetPort(MultiplyNode.kInputA).GetTypeField()) != GraphType.Height.One ||
+            GraphTypeHelpers.GetHeight(node.GetPort(MultiplyNode.kInputB).GetTypeField()) != GraphType.Height.One;
+
+            return new NodeUIDescriptor(
+                name: key.Name,
+                version: key.Version,
+                tooltip: "Calculates the value of input A multiplied by input B.",
+                categories: new string[] { "Math", "Basic" },
+                synonyms: new string[] { "multiplication", "*", "times", "x", "product" },
+                parameters: new ParameterUIDescriptor[] {
+                    new(name: MultiplyNode.kInputA, tooltip: "Input A"),
+                    new(name: MultiplyNode.kInputB, tooltip: "Input B"),
+                    new(name: MultiplyNode.kOutput,
+                    tooltip: isMatrixMultiplication ? "mul(A, B)" : "A * B")
+                },
+                displayName: isMatrixMultiplication ? "Multiply (Matrix)" : "Multiply (Component)"
```

Comment:
Not sure if we want something like this-- basically, the displayName can change to better reflect the underlying math operation that's happening.

Reply:
That's a cool idea, but I think that it may make the node too large.